src/Storages/MergeTree/MergeTreeMutationEntry.cpp

```diff
@@ -154,6 +167,20 @@ MergeTreeMutationEntry::MergeTreeMutationEntry(DiskPtr disk_, const String & pat
     }
 
     assertEOF(*buf);
+
+    for (const auto & command : commands)
```

Comment:
Please consider invoking `MergeTreeData::getPartitionIdsAffectedByCommands` here instead, since it looks identical to this piece of code